gov/types/codec.go

```diff
@@ -0,0 +1,42 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	authTypes "github.com/maticnetwork/heimdall/auth/types"
+)
+
+// ModuleCdc module codec
+var ModuleCdc = codec.New()
+
+// RegisterCodec registers all the necessary types and interfaces for
+// governance.
+func RegisterCodec(cdc *codec.Codec) {
+	cdc.RegisterInterface((*Content)(nil), nil)
+
+	cdc.RegisterConcrete(MsgSubmitProposal{}, "gov/MsgSubmitProposal", nil)
+	cdc.RegisterConcrete(MsgDeposit{}, "gov/MsgDeposit", nil)
+	cdc.RegisterConcrete(MsgVote{}, "gov/MsgVote", nil)
+
+	cdc.RegisterConcrete(TextProposal{}, "heimdall/TextProposal", nil)
```

Comment:
we are handling only ParameterChangeProposal

Please remove (SoftwareUpgradeProposal, TextProposal)

Reply:
Added.